File: commons-work-api/src/main/java/io/github/cooperlyt/mis/work/message/WorkCreateMessage.java
package io.github.cooperlyt.mis.work.message;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.io.Serial;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class WorkCreateMessage implements java.io.Serializable{

  @Serial
  private static final long serialVersionUID = 1L;

  private long workId;

  private Map<String,Object> data;
}

File: commons-work-api/src/main/java/io/github/cooperlyt/mis/work/message/WorkCreateMessage.kt
package io.github.cooperlyt.mis.work.message

import java.io.Serial

data class WorkCreateMessage(
    val define: String,
    val workId: Long,
    val tags: Set<String>,
    val isProcess: Boolean = false,
    val documentation: ProcessDocumentation? = null,
    val variables: Map<String, Any> = emptyMap(),
) : java.io.Serializable {

    companion object {
        @Serial
        private const val serialVersionUID: Long = 1L
    }
}